### DISCUSSION TITLE
Multi language JSON fields

### DISCUSSION THREAD

**Discussion by blondie63** [Laravel-Backpack/community-forum#441] [Q&A (Help)] [answered]
Is there any widget or plugin to handle multi-language json fields in a more elegant and practical way than the native backpack one?

Thanks for help

Mauro

**Comment by promatik** [comment]
Hi Mauro!
I think there isn't any packages for this.

By an "elegant way", do you mean to not store the data all in one table cell with a JSON?

I've been using the "native" way for years and I haven't found any issues with it. Many times I need to provide data to external apps with APIs, and by using Resources I think everything gets very well organized 👌

Actually, some long time ago I thought about creating some sort of package, to handle translations in a more elegant way, where they would be stored in a specific table with multiple rows for each locale. But honestly, in my opinion, having the cell with all translations encoded in a JSON may actually be cleaner than having infinite rows in a translation table 🤷‍♂️ — no joins needed, no foreign key dependencies ...
_This is obviously just my opinion!_

**Reply by blondie63** [reply]
Hi Promatik,

I also think that storing translations in a single json field is very convenient and works well in many cases, what I don't like is the display and modification system.  I prefer to see and change the various languages ​​in the same screen without using a language switcher

**Reply by promatik** [reply]
Oh I see, you're talking about something like this; https://github.com/Laravel-Backpack/community-forum/issues/373

Also, please check what's on our plan; https://github.com/Laravel-Backpack/CRUD/pull/4267

**Comment by promatik** [comment, marked as answer]
Duplicate of https://github.com/Laravel-Backpack/community-forum/issues/373
